### PR TITLE
feat(docs): add inline documentation to universe functions m-r

### PR DESCRIPTION
**Pull request by sanderson**
Adds inline documentation for functions in the universe package (m-r).

### Done checklist
- [x] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

File: stdlib/universe/universe.flux [hunk 1]
```diff
@@ -1541,22 +1810,293 @@ builtin quantile : (
...
     where
     A: Record,
     B: Record
+
+// range filters rows based on time bounds.
+//
+// Input data must have a `_time` column of type time.
+// Rows with a null value in the `_time` are filtered.
+// Each input table’s group key value is modified to fit within the time bounds.
+// Tables with all rows outside the time bounds are filtered entirely.
```
Comment (wolffcm) [inline stdlib/universe/universe.flux:1899]: It's probably worth mentioning here that this transformation will add `_start` and `_stop` columns to its output, and also add them to the group key.

**Review by wolffcm — APPROVED** [review]
Looks good, I found a few typos and had some suggestions.

File: stdlib/universe/universe.flux [hunk 2]
```diff
@@ -1525,12 +1525,281 @@ builtin last : (<-tables: [A], ?column: string) => [A] where A: Record
...
+// - unsigned integer
+// - string
+// - boolean
+// - time
+//
+// ### Output data
+// Output tables are the result of applying the map function (`fn`) to each
+// record of the input tables. Output records are assigned to new tables based
+// on the group key of the input stream.
+// If output record contains a different value for a group key column, the
```
Comment (noramullen1) [inline stdlib/universe/universe.flux:1546]:
```suggestion
// If the output record contains a different value for a group key column, the
```

File: stdlib/universe/universe.flux [hunk 3]
```diff
@@ -1541,22 +1810,293 @@ builtin quantile : (
...
+// - The label of a new column is the concatenation of the values of `columnKey`
+//   using `_` as a separator. If the value is null, "null" is used.
+// - A new row is created for each unique value identified by the
+//   `rowKey` parameter.
+// - For each new row, values for group key columns stay the same, while values
+//   for new columns are determined from the input tables by the value in
+//   `valueColumn` at the row identified by the `rowKey` values and the new
+//   column’s label. If no value is found, the value is set to `null`.
+// - Any column that is not part of the group key or not specified in the
+//   `rowKey`, `columnKey` and `valueColumn` parameters is dropped.
```
Comment (noramullen1) [inline stdlib/universe/universe.flux:1843]:
```suggestion
//   `rowKey`, `columnKey`, and `valueColumn` parameters is dropped.
```

**Review by noramullen1 — APPROVED** [review]

File: stdlib/universe/universe.flux [hunk 4]
```diff
@@ -1525,12 +1525,281 @@ builtin last : (<-tables: [A], ?column: string) => [A] where A: Record
...
+// If output record contains a different value for a group key column, the
+// record is regrouped into the appropriate table.
+// If the output record drops a group key column, that column is removed from
+// the group key.
+//
+// #### Preserve columns
+// By default, map() drops any columns that:
+//
+// - Are not part of the input table’s group key.
+// - Are not explicitly mapped in the `fn` function.
```
[the remark this answers is not in the document]
Comment (sanderson) [reply stdlib/universe/universe.flux]: Thanks, I updated this section a bit.

File: stdlib/universe/universe.flux (hunk 1 above)
Comment (sanderson) [reply stdlib/universe/universe.flux:1896]: Good call. Added this information in.

**Pull request merged**